prover/lang/kore-lang.md

```diff
@@ -565,14 +439,17 @@ where the term being unfolded has been replace by `#hole`.
     => \member(subst(LHS, X, V), subst(RHS, X, V))
   rule subst(\subseteq(LHS, RHS):Pattern, X, V)
     => \subseteq(subst(LHS, X, V), subst(RHS, X, V))
-  rule subst(\forall { E } C, X, V) => \forall { E } C
-    requires X in E orBool X in PatternsToVariableNameSet(E)
-  rule subst(\forall { E } C, X, V) => \forall { E } subst(C, X, V)
-    requires notBool( X in E orBool X in PatternsToVariableNameSet(E) )
-  rule subst(\exists { E } C, X, V) => \exists { E } C
-    requires X in E orBool X in PatternsToVariableNameSet(E)
-  rule subst(\exists { E } C, X, V) => \exists { E } subst(C, X, V)
-    requires notBool( X in E orBool X in PatternsToVariableNameSet(E) )
+
+  rule subst(\forall { E } C, X, V) => \forall { E } C requires X in E
+  rule subst(\forall { E } C, X, V) => \forall { E } C requires X in PatternsToVariableNameSet(E)
+  rule subst(\forall { E } C, X, V) => \forall { E } subst(C, X, V) requires notBool( X in E )
+  rule subst(\forall { E } C, X, V) => \forall { E } subst(C, X, V) requires notBool( X in PatternsToVariableNameSet(E) )
+
+  rule subst(\exists { E } C, X, V) => \exists { E } C requires X in E
+  rule subst(\exists { E } C, X, V) => \exists { E } C requires X in PatternsToVariableNameSet(E)
+  rule subst(\exists { E } C, X, V) => \exists { E } subst(C, X, V) requires notBool( X in E )
+  rule subst(\exists { E } C, X, V) => \exists { E } subst(C, X, V) requires notBool( X in PatternsToVariableNameSet(E) )
```

Comment:
This is suspicious. What if `X in E` and `notBool(X in PatternsToVariableNameSet(E))` at the same time? And what if `notBool(X in E)`. and `X in PatternsToVariableNameSet(E)` at the sime time? Why do we need this refactoring anyway?